src/graphics/components/Sprite.cpp

```diff
@@ -23,6 +24,11 @@ Sprite::Sprite(IObject *owner, const std::string &path) : AComponent(
 
 Sprite::Sprite(IObject *owner, const json::JsonObject *data) : AComponent(
     owner, new Meta(this), data) {
+    this->deserializeFields(data);
+    if (!texture.loadFromFile(path)) {
+        throw GraphicsException("Failed to load texture from file: " + std::string(path));
+    }
```

Comment:
I think the ` // Todo: add sample texture and only log error` can be put here too